Orange/widgets/utils/state_summary.py

```diff
@@ -57,10 +57,47 @@ def _plural(number):
         targets = format_variables_string(data.domain.class_vars)
         metas = format_variables_string(data.domain.metas)
 
+        features_missing = missing_values(data.has_missing_attribute()
+                                          and data.get_nan_frequency_attribute())
         n_features = len(data.domain.variables) + len(data.domain.metas)
         details = \
             f'{len(data)} instance{_plural(len(data))}, ' \
             f'{n_features} variable{_plural(n_features)}\n' \
-            f'Features: {features}\nTarget: {targets}\nMetas: {metas}'
-
+            f'Features: {features} {features_missing}\n' \
+            f'Target: {targets}\nMetas: {metas}'
     return details
+
+
+def missing_values(value):
+
+    if value:
+        return f'({value*100:.1f}% missing values)'
+    else:
+        return f'(No missing values)'
+
+
+def format_multiple_input(inputs):
+    """
+    A function that forms the entire descriptive part of the input/output
+    summary for widgets that have more than one input.
+
+    :param inputs: A list of tuples for each input dataset where the first
+    element of the tuple is the name of the dataset and the second is the
+    dataset
+    :type inputs: list(tuple(str, Orange.data.Table))
+
+    :return: A formatted string
+    """
+
```

Comment:
Inputs where dataset is None are insufficient handled. E.g.: formatted string for a widget with i.e. three inputs where two of those are None, results in horizontal lines with no text in between.

<img width="354" alt="Screenshot 2020-03-20 at 14 31 48" src="https://user-images.githubusercontent.com/11465003/77168163-a4859a00-6ab7-11ea-8083-d33bccdd784e.png">

I'd recommend introducing default strings for "None" datasets (i.e. "No data on input" for inputs and "No data on output" for outputs).

Reply:
Didn't we say that when there is no data on output there will be just output icon with no summary and/or detail? Or did you maybe mean that we could implement `format_multiple_inputs` function into widgets that have multiple outputs and the user can see data info for all outputs of the widget?

Reply:
I meant for the widgets with multiple equivalent inputs. If all of them are None, just the icon is seen, without the summary and detail. But if one is not None and two are, then we get something similar as above.